drivers/bluetooth/hci/hci_silabs_siwx91x.c

```diff
@@ -5,21 +5,68 @@
 
 #include <zephyr/kernel.h>
 #include <zephyr/drivers/bluetooth.h>
+#include <zephyr/sys/byteorder.h>
 
 #define DT_DRV_COMPAT silabs_siwx91x_bt_hci
 #define LOG_LEVEL     CONFIG_BT_HCI_DRIVER_LOG_LEVEL
 #include <zephyr/logging/log.h>
 LOG_MODULE_REGISTER(bt_hci_driver_siwg917);
 
 #include "rsi_ble.h"
+#include "rsi_ble_common_config.h"
+#define BLE_RF_POWER_INDEX     0x0006
+#define BT_OP_VS_RF_POWER_MODE BT_OP(BT_OGF_VS, BLE_RF_POWER_INDEX)
+#define BT_LE_MODE             2
 
+static int rsi_bt_driver_send_tx_pwr_vs_cmd(const struct device *dev);
 static void siwx91x_bt_resp_rcvd(uint16_t status, rsi_ble_event_rcp_rcvd_info_t *resp_buf);
 
 struct hci_data {
 	bt_hci_recv_t recv;
 	rsi_data_packet_t rsi_data_packet;
 };
 
+/**
+ * @brief Send RF power mode configuration command to controller
+ * @param dev Pointer to the device structure
+ * @return 0 on success, negative errno on failure
+ */
+static int rsi_bt_driver_send_tx_pwr_vs_cmd(const struct device *dev)
+{
+	struct net_buf *buf;
+	uint8_t le_tx_power_index = RSI_BLE_PWR_INX;
+	uint8_t protocol_mode = BT_LE_MODE;
+	int err;
+
+	/* Allocate HCI command buffer with timeout */
+	buf = bt_hci_cmd_alloc(K_FOREVER);
+	if (!buf) {
+		LOG_ERR("Failed to allocate HCI command buffer");
+		return -ENOMEM;
+	}
+
+	/* Manually add HCI command header */
+	net_buf_add_le16(buf, BT_OP_VS_RF_POWER_MODE);
+	net_buf_add_u8(buf, sizeof(protocol_mode) + sizeof(le_tx_power_index));
```

Comment:
This is not correct. `bt_hci_cmd_send_sync()` will insert these automatically into the buffer. Did you actually test this?

Reply:
It is still not correct. The controller will be receiving invalid (or at least unintended) values for the parameters, since it'll interpret the above bytes as parameters instead of the command header.